client/cli/src/runtime.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+use futures::{Future, future, future::FutureExt};
+use futures::select;
+use futures::pin_mut;
+use sc_service::{AbstractService, Configuration};
+use crate::error;
+use crate::informant;
+
+async fn main<F, E>(func: F) -> Result<(), Box<dyn std::error::Error>>
+where
+	F: Future<Output = Result<(), E>> + future::FusedFuture,
+	E: 'static + std::error::Error,
+{
+	use tokio::signal::unix::{signal, SignalKind};
+
+	let mut stream_int = signal(SignalKind::interrupt())?;
+	let mut stream_term = signal(SignalKind::terminate())?;
+
+	let t1 = stream_int.recv().fuse();
+	let t2 = stream_term.recv().fuse();
+	let t3 = func;
+
+	pin_mut!(t1, t2, t3);
+
+	select! {
+		_ = t1 => println!("Caught SIGINT"),
```

Comment:
You want eprintln?

Reply:
More like: why is anything printed at all?

Reply:
Because usually that's what you do when you catch a signal?

Anyway. Do I remove it then?

Reply:
Personally I prefer the messages that mednafen displays for SIGTERM and SIGINT:

* SIGTERM: "MUST TERMINATE ALL HUMANS"
* SIGINT: "How dare you interrupt me!"

But really I want to move on because I'm getting new conflicts every day. So I can remove it if that's what you want.

Reply:
New merge conflicts... can we finish this review? :pray:

Reply:
It's really fine as it is now. If you think it's a bad decision just tell me what to do exactly and I will change it.

Reply:
Delete it please.

Reply:
done